mangaki/mangaki/utils/mal.py

```diff
@@ -220,18 +229,72 @@ def search_works(self, work_type: MALWorks, query: str) -> List[MALEntry]:
                    getattr(settings, 'MAL_PASS', None))
 
 
-def insert_into_mangaki_database_from_mal(mal_entries: List[MALEntry]):
+def look_for_works(work_list: QuerySet,
+                   ext_poster: str,
+                   titles: List[str]) -> List[Work]:
+    constraints_work = None
+    constraints_title = None
+    for synonym in titles:
+        cons = SearchQuery(synonym, config='simple')
+
+        if constraints_work:
+            constraints_work |= cons
+        else:
+            constraints_work = cons
+
+        if constraints_title:
+            constraints_title |= cons
+        else:
+            constraints_title = cons
+
+    constraints_work = Q(title_search=constraints_work) | Q(ext_poster=ext_poster)
+
+    works = set(work_list.filter(constraints_work)[:2])
+    titles = list(
+        WorkTitle.objects
+            .filter(title_search=constraints_title)
+            .values_list('work_id', flat=True).all()
+    )
+    more_works = set(Work.objects.in_bulk(titles))
```

Comment:
(In that case `works` and `titles` should probably be renamed something like `ids_work_matched` and `ids_title_matched`)